Services/DeviceClient.cs

```diff
@@ -155,6 +155,8 @@ public Task UpdateTwinAsync(Device device)
                         Exception = e.GetType().FullName,
                         e.InnerException
                     });
+
+                throw new TelemetrySendException("Message delivery failed with " + e.Message);
```

Comment:
todo: create a github issue to add more granular checks here, i.e. we need to handle different errors in different ways.

Reply:
also, pass the original exception back
```
throw new TelemetrySendException("Message delivery failed with " + e.Message, e);
```